src/components/tooltip/TooltipDirective.js

```diff
@@ -118,7 +118,10 @@ goog.require('ga_topic_service');
                 updateCursorStyle, 10, false, false);
 
         // Register click/touch/mousemove events on map
-        var registerMapEvents = function(scope, onClick) {
+        var registerMapEvents = function(scope, onClick, deregEvent) {
```

Comment:
Hmm, `dereg = angular.noop` is only set if somebody actually calls the `deregMapEvents` function (and not when calling the `registerMapEvents`.

Anyway, we could use a module-wide variably by moving `var deregMapEvents = angular.noop` out of link function and into module function and then use this variable always. We always set `deregMapEvents` either inside the register functions or inside the deregister function.

```
var registerMapEvents = function(scope, onClick) {
        if (deregMapEvents != angular.noop) {
            return;
        }
        .
        .
        .
        deregMapEvents = function() {
            deregMapClick();
            ol.Observable.unByKey(deregPointerMove);
            deregMapEvents = angular.noop;
        }
        // Nothing to return, as deregMapEvents is outside of this function scope.
    };
```

Working?